bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBaseUnitHandler.java

```diff
@@ -0,0 +1,154 @@
+
+    protected final Logger logger = LoggerFactory.getLogger(TouchWandBaseUnitHandler.class);
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(
+            Arrays.asList(THING_TYPE_SHUTTER, THING_TYPE_SWITCH, THING_TYPE_WALLCONTROLLER, THING_TYPE_DIMMER));
+
+    @NonNullByDefault({})
+    protected String unitId;
+
+    @NonNullByDefault({})
+    protected TouchWandBridgeHandler bridgeHandler;
+
+    public TouchWandBaseUnitHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // updateTouchWandUnitState(getUnitState(unitId));
+        } else {
+            touchWandUnitHandleCommand(command);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (bridgeHandler != null) {
+            bridgeHandler.unregisterUpdateListener(this);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge == null || !(bridge.getHandler() instanceof TouchWandBridgeHandler)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            logger.warn("Trying to initialize {} without a bridge", getThing().getUID());
+            return;
+        }
+
+        bridgeHandler = (TouchWandBridgeHandler) bridge.getHandler();
+
+        unitId = getThing().getUID().getId(); // TouchWand unit id
+
+        bridgeHandler.registerUpdateListener(this);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            boolean thingReachable = false;
+            String response = bridgeHandler.touchWandClient.cmdGetUnitById(unitId);
+            thingReachable = !response.isEmpty();
+            if (thingReachable) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        });
+    }
+
+    private int getUnitState(String unitId) {
+        int status = 0;
+
+        if (bridgeHandler == null) {
+            return status;
+        }
+
+        String response = bridgeHandler.touchWandClient.cmdGetUnitById(unitId);
+        if (!response.isEmpty()) {
+            return status;
+        }
+
+        JsonParser jsonParser = new JsonParser();
+
+        try {
+            JsonObject unitObj = jsonParser.parse(response).getAsJsonObject();
+            status = unitObj.get("currStatus").getAsInt();
+            if (!this.getThing().getStatusInfo().getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (JsonParseException | IllegalStateException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Could not parse cmdGetUnitById");
```

Comment:
You could include the exception's message.
```suggestion
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Could not parse cmdGetUnitById: "+e.getMessage());
```